src/main/java/com/quizapp/enterprise/services/GameService.java

```diff
@@ -25,7 +25,12 @@
     @Autowired private GameOverEventPublisher gameOverEventPublisher;
 
     @Override
-    public Game startNewGame(int quizId) {
+    public Game startNewGame(int quizId) throws Exception {
+
+        ArrayList<Question> questions = (ArrayList<Question>) questionRepository.findByquizId(quizId);
+        if(questions == null){
+            throw new Exception("quiz not found");
+        }
```

Comment:
included a check for a possible null prior to execution on possible null object so code may handle the case as desired.
This change was recommended by the "8. Proper handling of Null Pointer Exceptions"  section of  the java coding best practices summary referenced below

source:https://xperti.io/blogs/java-coding-best-practices/

Reply:
See notes above on null check.